HW10: Dmitry Kolyago

dzmitrysk opened the pull request:
### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8).
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

dzmitrysk committed:
HW10: Dmitry Kloyago

dzmitrysk committed:
HW10: Dmitry Kloyago

dzmitrysk committed:
Merge remote-tracking branch 'origin/Kolyago-10' into Kolyago-10

# Conflicts:
#	src/task_10_02.py

dzmitrysk committed:
Fixed flake8 remarks

dzmitrysk committed:
Fixed flake8 remarks

dzmitrysk committed:
Merge branch 'master' into Kolyago-10

dzmitrysk committed:
Delete data_01.csv

dzmitrysk committed:
Delete data_weather.csv

dzmitrysk committed:
Delete datas.csv

dzmitrysk committed:
Update data files

dzmitrysk committed:
Merge remote-tracking branch 'origin/Kolyago-10' into Kolyago-10

# Conflicts:
#	src/data_01.csv
#	src/data_weather.csv
#	src/datas.csv

dzmitrysk committed:
Update data files

dzmitrysk committed:
Changed lineterminator for .csv: from "\r" to "\n". Updated data files

tamkovich commented on src/task_10_01.py:
```diff
@@ -0,0 +1,33 @@
+peoples_in_age_ranges = {"1-12": 0, "13-18": 0, "19-25": 0, "26-40": 0, "40+": 0}
+
+for row in data:
+    if int(row[2]) <= 12:
```
сделай перевод числа в int перед иф

dzmitrysk replied:
Applied

tamkovich commented on src/task_10_01.py:
```diff
@@ -0,0 +1,33 @@
+data = []
+with open("data_01.csv", "r") as fin:
+    f_reader = csv.reader(fin, delimiter=",")
+    for row in f_reader:
+        data.append(row)
```
сделай через list comprehensions

dzmitrysk replied:
Applied

tamkovich commented on src/task_10_01.py:
```diff
@@ -0,0 +1,33 @@
+    f_reader = csv.reader(fin, delimiter=",")
+    for row in f_reader:
+        data.append(row)
+del(data[0])
```
```suggestion
del data[0]
```

dzmitrysk replied:
Applied

tamkovich commented on src/task_10_02.py:
```diff
@@ -0,0 +1,38 @@
+        data.append(row)
+del (data[0])
+
+weather_minsk = list(filter(lambda row: row if row[1] == "Minsk" else None, data))
```
```suggestion
weather_minsk = list(filter(lambda row: row[1] == "Minsk", data))
```

dzmitrysk replied:
Applied

tamkovich commented on src/task_10_02.py:
```diff
@@ -0,0 +1,38 @@
+    f_reader = csv.reader(fin, delimiter=",")
+    for row in f_reader:
+        data.append(row)
+del (data[0])
```
```suggestion
del data[0]
```

dzmitrysk replied:
Applied

tamkovich commented on src/task_10_02.py:
```diff
@@ -0,0 +1,38 @@
+data = []
+with open("data_weather.csv", "r") as fin:
+    f_reader = csv.reader(fin, delimiter=",")
+    for row in f_reader:
+        data.append(row)
```
please use list comprehensions

dzmitrysk replied:
Applied

tamkovich commented on src/task_10_02.py:
```diff
@@ -0,0 +1,38 @@
+weather_minsk = list(filter(lambda row: row if row[1] == "Minsk" else None, data))
+weather_minsk = list(
+    filter(
+        lambda row: row if today - seek_period <= datetime.strptime(row[0], '%Y-%m-%d') <= today
+        else None, weather_minsk))
```
```suggestion
        lambda row: today - seek_period <= datetime.strptime(row[0], '%Y-%m-%d') <= today, weather_minsk))
```

dzmitrysk replied:
Applied

tamkovich commented on src/task_10_03.py:
```diff
@@ -0,0 +1,26 @@
+datas = []
+with open("datas.csv", "r") as fin:
+    f_reader = csv.reader(fin, delimiter=",")
+    for row in f_reader:
+        datas.append(row)
```
pls use list comprehensions

dzmitrysk replied:
Applied

tamkovich commented on src/task_10_03.py:
```diff
@@ -0,0 +1,26 @@
+    f_reader = csv.reader(fin, delimiter=",")
+    for row in f_reader:
+        datas.append(row)
+del(datas[0])
```
```suggestion
del datas[0]
```

dzmitrysk replied:
Applied

dzmitrysk committed:
Changes on Tamkovich code review

1) Applied list comprehensions on reading from .csv
2) Simplified lambda fn on using filter()
3) Removed redundant brackets on using "del"

dzmitrysk committed:
Changes on Tamkovich code review

1) Applied list comprehensions on reading from .csv
2) Simplified lambda fn on using filter()
3) Removed redundant brackets on using "del"

dzmitrysk committed:
Merge remote-tracking branch 'origin/Kolyago-10' into Kolyago-10

# Conflicts:
#	src/task_10_02.py

tamkovich commented on src/task_10_01.py:
```diff
@@ -0,0 +1,33 @@
+    age = int(row[2])
+    if age <= 12:
+        peoples_in_age_ranges["1-12"] += 1
+    elif 12 < age <= 18:
```
И везде убери лишние проверки

```suggestion
    elif age <= 18:
```

dzmitrysk replied:
Applied

tamkovich reviewed (dismissed):
LGTM, но пофикси последний коммент

dzmitrysk committed:
Exclude redundant condition

tamkovich commented on src/task_10_01.py:
```diff
@@ -0,0 +1,33 @@
+        peoples_in_age_ranges["1-12"] += 1
+    elif age <= 18:
+        peoples_in_age_ranges["13-18"] += 1
+    elif 18 < age <= 25:
```
:)
```suggestion
    elif age <= 25:
```

tamkovich commented on src/task_10_01.py:
```diff
@@ -0,0 +1,33 @@
+        peoples_in_age_ranges["13-18"] += 1
+    elif 18 < age <= 25:
+        peoples_in_age_ranges["19-25"] += 1
+    elif 25 < age <= 40:
```
```suggestion
    elif age <= 40:
```

tamkovich reviewed (changes requested):
.

dzmitrysk committed:
Update src/task_10_01.py

Co-authored-by: Denis Tamkovich <35404746+tamkovich@users.noreply.github.com>

dzmitrysk committed:
Update src/task_10_01.py

Co-authored-by: Denis Tamkovich <35404746+tamkovich@users.noreply.github.com>
